animation_nodes/algorithms/mesh/points_scatter.pyx

```diff
@@ -113,7 +132,73 @@ cdef Vector3DList sampleRandomPoints(Vector3DList vertices, PolygonIndicesList p
             v.x = p1 * v1.x + p2 * v2.x + p3 * v3.x
             v.y = p1 * v1.y + p2 * v2.y + p3 * v3.y
             v.z = p1 * v1.z + p2 * v2.z + p3 * v3.z
-            points.data[index] = v
+
+            directionToMatrix_LowLevel(matrices.data + index, &direction, &guide, 2, 0)
+            setMatrixTranslation(matrices.data + index, &v)
```

Comment:
Use the `createMatrix` function here. You already have the normal and the tangent (The guide). Normalize them both, compute the bitangent by taking the cross product of the normalized normal and tangent, then pass them to the `createMatrix` function.

Reply:
Omar, should I call that function from `mesh/c_utils.pyx` or define it here?

Reply:
Move it somewhere common like `math/matrix`.

Reply:
Okay.

Reply:
Also, perhaps name it something more approperiate like `matrixFromNormalizedAxisData`.